mirrord/layer/src/socket/ops.rs

```diff
@@ -711,23 +713,24 @@ pub(super) fn getaddrinfo(
 fn remote_hostname_string() -> Detour<CString> {
     let hostname_path = PathBuf::from("/etc/hostname");
 
-    let hostname_fd = file::ops::open(
-        Detour::Success(hostname_path),
+    let OpenFileResponse { fd } = file::ops::RemoteFile::remote_open(
+        hostname_path,
         OpenOptionsInternal {
             read: true,
             ..Default::default()
         },
     )?;
 
-    let hostname_file = file::ops::read(hostname_fd, 256)?;
+    let ReadFileResponse { bytes, read_amount } = file::ops::RemoteFile::remote_read(fd, 256)?;
 
-    close_layer_fd(hostname_fd);
+    let _ = file::ops::RemoteFile::remote_close(fd).inspect_err(|fail| {
```

Comment:
Don't you still need to close the layer fd?

Reply:
We should be opening the file now only in the remote, no local fd is created, and we don't insert the file into `OPEN_FILES` either, so there is nothing local to close.

Reply:
Gotcha